src/orchestrator.ts

```diff
@@ -167,11 +159,14 @@ export class Orchestrator {
                     return;
                 }
 
-                const newDecisionStartedEvent = state.find(
-                    (e) =>
-                        e.EventType === HistoryEventType.OrchestratorStarted &&
-                        e.Timestamp > decisionStartedEvent.Timestamp
-                );
+                const newDecisionStartedEvent = state
+                    .filter(
+                        (e, index) =>
+                            e.EventType === HistoryEventType.OrchestratorStarted &&
+                            e.Timestamp > decisionStartedEvent.Timestamp &&
+                            index < completionIndex
+                    )
+                    .pop();

```

Comment:
What's the `.pop()` for? Do we normally change the state (history) inside orchestrator functions? This seems strange to me because I would assume the state should be immutable here.

Reply:
If I understand this correctly, it won't edit state. `.filter()` creates a copy of the list.

This is equivalent to a `.Select().Last()` LINQ query in C#.

Reply:
No objection if this is just operating on a copy of the history. 👍

Reply:
This is precisely right ;)